### PR TITLE
Fixed #26747 -- Used more specific assertions in the Django test suite.

### PR DESCRIPTION
**Pull request by jdufresne**
<https://code.djangoproject.com/ticket/26747>

### CODE REVIEW

File: docs/intro/tutorial05.txt [hunk 1]
```diff
@@ -180,7 +180,7 @@ Put the following in the ``tests.py`` file in the ``polls`` application:
             """
             time = timezone.now() + datetime.timedelta(days=30)
             future_question = Question(pub_date=time)
-            self.assertEqual(future_question.was_published_recently(), False)
+            self.assertFalse(future_question.was_published_recently())
```
Comment (timgraham) [inline docs/intro/tutorial05.txt]: I'm not convinced about the merits of `assertTrue`/`False` since such assertions pass if `bool(val)` is `True`/`False`.

**Comment by jdufresne** [comment]
> I'm not convinced about the merits of assertTrue/False since such assertions pass if bool(val) is True/False.

Makes sense.

Do you think these should be changed to `assertIs()` as [recommended by the documentation](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue), or left as is?

**Comment by jdufresne** [comment]
I did find the pattern:

```
self.assertEqual(bool(p.mugshot), False)
```

Asserts that follow this pattern have been modified to use `assertFalse()`.

**Comment by timgraham** [comment]
Yea, I guess `assertIs` is preferred.

**Comment by jdufresne** [comment]
Made discussed changes.

File: docs/intro/tutorial05.txt [hunk 2]
```diff
@@ -203,7 +203,7 @@ and you'll see something like::
     ----------------------------------------------------------------------
     Traceback (most recent call last):
       File "/path/to/mysite/polls/tests.py", line 16, in test_was_published_recently_with_future_question
-        self.assertEqual(future_question.was_published_recently(), False)
+        self.assertIs(future_question.was_published_recently(), False)
     AssertionError: True != False
```
Comment (timgraham) [inline docs/intro/tutorial05.txt]: The message will change here: `True is not False`

**Comment by jdufresne** [comment]
Made changes. All additional feedback welcome. Thanks.

File: tests/model_fields/test_imagefield.py [hunk 3]
```diff
@@ -120,18 +120,18 @@ def test_equal_notequal_hash(self):
         p1.mugshot.save("mug", self.file1)
         p2 = self.PersonModel(name="Bob")
         p2.mugshot.save("mug", self.file2)
-        self.assertEqual(p1.mugshot == p2.mugshot, False)
-        self.assertEqual(p1.mugshot != p2.mugshot, True)
+        self.assertIs(p1.mugshot == p2.mugshot, False)
+        self.assertNotEqual(p1.mugshot, p2.mugshot)
```
Comment (timgraham) [inline tests/model_fields/test_imagefield.py]: assertIs here and below?

**Comment by jdufresne** [comment]
> assertIs here and below?

Fixed.

**Pull request merged**